### PR TITLE
Add more doc for binary blds and annotations

### LINKED ISSUE #510: doc for the deployment exporter should explain how to label all "special" builds e.g. binary for example.
Recreate:
oc create namespace binary-build
oc new-build python --name=python-binary-build4 --binary=true -n binary-build
oc label bc python-binary-build4 app.kubernetes.io/name=python-binary-build4 -n binary-build
oc start-build bc/python-binary-build4 --from-file=./app.py --follow -n binary-build
oc get builds -n binary-build
oc -n binary-build annotate build python-binary-build4-1 --overwrite \
io.openshift.build.commit.id=7810f2a85d5c89cb4b17e9a3208a311af65338d8 \
io.openshift.build.source-location=http://github.com/konveyor/pelorus
oc -n binary-build new-app python-binary-build4

* The build has labels:
```
app.kubernetes.io/name=python-binary-build4
build=python-binary-build4
buildconfig=python-binary-build4
openshift.io/build-config.name=python-binary-build4
openshift.io/build.start-policy=Serial
```

This is likely due to:
https://github.com/konveyor/pelorus/blob/master/exporters/deploytime/app.py#L218

```
[whayutin@thinkdoe BUILD]$ oc get all
NAME                                        READY   STATUS      RESTARTS   AGE
pod/python-binary-build4-1-build            0/1     Completed   0          4m15s
pod/python-binary-build4-5747676745-68drq   1/1     Running     0          2m51s

NAME                           TYPE        CLUSTER-IP       EXTERNAL-IP   PORT(S)    AGE
service/python-binary-build4   ClusterIP   172.30.197.209   <none>        8080/TCP   2m52s

NAME                                   READY   UP-TO-DATE   AVAILABLE   AGE
deployment.apps/python-binary-build4   1/1     1            1           2m52s

NAME                                              DESIRED   CURRENT   READY   AGE
replicaset.apps/python-binary-build4-5747676745   1         1         1       2m51s
replicaset.apps/python-binary-build4-5f575c5f87   0         0         0       2m52s

NAME                                                  TYPE     FROM     LATEST
buildconfig.build.openshift.io/python-binary-build4   Source   Binary   1

NAME                                              TYPE     FROM     STATUS     STARTED         DURATION
build.build.openshift.io/python-binary-build4-1   Source   Binary   Complete   4 minutes ago   54s

NAME                                                  IMAGE REPOSITORY                                                                     TAGS     UPDATED
imagestream.image.openshift.io/python-binary-build4   image-registry.openshift-image-registry.svc:5000/binary-build/python-binary-build4   latest   3 minutes ago
[whayutin@thinkdoe BUILD]$ 

```

### PR DESCRIPTION
**Pull request by weshayutin**
* seperate pelorus config into "overview" and "details"
* add more concrete examples for annotating a binary build
* point to the tekton demo as a doc resource for supporting
various build incantations.

## Linked Issues?

resolves #510

### CODE REVIEW

File: docs/Configuration.md [hunk 1]
```diff
@@ -263,6 +254,39 @@ Custom Annotation names may also be configured using ConfigMap Data Values.
...
+#### An example workflow for an OpenShift binary build:
+
+```
+NS=mongo-persistent
+NAME=python-binary-build
+
+oc create namespace "${NS}"
+
+oc new-build python --name="${NAME}" --binary=true -n "${NS}"  -l "app.kubernetes.io/name=${NAME}"
+oc start-build "bc/${NAME}" --from-file=./app.py --follow -n "${NS}"
```
Comment (mpryc) [inline docs/Configuration.md:266]: First step is to ensure `app.py` is in the local folder from which build commands are executed.

File: docs/Configuration.md [hunk 2]
```diff
@@ -263,6 +254,39 @@ Custom Annotation names may also be configured using ConfigMap Data Values.
...
+
+
+oc -n "${NS}" annotate build "${NAME}-1" --overwrite \
```
Comment (mpryc) [inline docs/Configuration.md:269]: prior to that command ensure build is created:

```
$ oc get builds -n "${NS}"
NAME                  TYPE      FROM          STATUS     STARTED        DURATION
python-binary-build-1   Source    Binary    Complete   53 seconds ago   50s

```

**Review by scoonRH — APPROVED** [review]

File: docs/Configuration.md (hunk 1 above)
Comment (weshayutin) [reply docs/Configuration.md:271]: thanks, fixed

File: docs/Configuration.md (hunk 2 above)
Comment (weshayutin) [reply docs/Configuration.md:274]: thanks, fixed

File: docs/Configuration.md [hunk 3]
```diff
@@ -263,6 +254,44 @@ Custom Annotation names may also be configured using ConfigMap Data Values.
 
 Note: The requirement to label the build with `app.kubernetes.io/name=<app_name>` for the annotated Builds applies.
 
+#### An example workflow for an OpenShift binary build:
+
+```
+# sample_app in current directory
+cat app.py 
+#!/usr/bin/env python3
+print("Hello World")
```
Comment (KevinMGranger) [inline docs/Configuration.md:263]: Can we make this two different snippets to reduce confusion?

<pre>
```python
# app.py contents here
```
</pre>
<pre>
```bash
# script contents here
```
</pre>

Comment (weshayutin) [reply docs/Configuration.md]: aye.. thank you!

**Pull request merged**